server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCicsRun.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.usecases;
+
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.common.error.ErrorSource;
+import org.eclipse.lsp.cobol.usecases.common.CICSTestUtils;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+/**
+ * Test CICS RUN command. Documentation link: <a
+ * href="https://www.ibm.com/docs/en/cics-ts/6.x?topic=summary-run">RUN
+ * Command</a>
+ *
+ * <p>This class tests all variations of the RUN command found in the link above.
+ */
+public class TestCicsRun {
+    private static final String DEFAULT_ALL_VALID_ONE = "RUN ACQACTIVITY ASYNCHRONOUS INPUTEVENT(12) FACILITYTOKN(123)";
+    private static final String DEFAULT_ALL_VALID_TWO = "RUN ACQPROCESS INPUTEVENT(2) SYNCHRONOUS";
+    private static final String DEFAULT_ALL_VALID_THREE = "RUN SYNCHRONOUS ACTIVITY(12) INPUTEVENT(12)";
+    private static final String DEFAULT_ALL_VALID_FOUR = "RUN INPUTEVENT(12) ASYNCHRONOUS FACILITYTOKN(123) ACTIVITY(12)";
+    private static final String DEFAULT_ALL_VALID_FIVE = "RUN FACILITYTOKN(123) ACQACTIVITY INPUTEVENT(12) ASYNCHRONOUS";
+
+    private static final String DEFAULT_SOME_VALID_ONE = "RUN ASYNCHRONOUS ACTIVITY(12) INPUTEVENT(12)";
+    private static final String DEFAULT_SOME_VALID_TWO = "RUN ASYNCHRONOUS FACILITYTOKN(123) ACQACTIVITY";
+    private static final String DEFAULT_SOME_VALID_THREE = "RUN ACQPROCESS ASYNCHRONOUS INPUTEVENT(12)";
+
+    private static final String DEFAULT_BARE_VALID_ONE = "RUN SYNCHRONOUS ACQPROCESS";
+    private static final String DEFAULT_BARE_VALID_TWO = "RUN ACQACTIVITY ASYNCHRONOUS";
+    private static final String DEFAULT_BARE_VALID_THREE = "RUN SYNCHRONOUS ACQACTIVITY";
+    private static final String DEFAULT_BARE_VALID_FOUR = "RUN FACILITYTOKN(123) ACQACTIVITY";
+
+    private static final String DEFAULT_INVALID_ONE = "RUN SYNCHRONOUS ACTIVITY(12) {FACILITYTOKN|error1}(123)";
+    private static final String DEFAULT_INVALID_TWO = "RUN ASYNCHRONOUS {ACQPROCESS|error1}  {ACQACTIVITY|error1} INPUTEVENT(12)";
+    private static final String DEFAULT_INVALID_THREE = "RUN ACTIVITY(12) SYNCHRONOUS INPUTEVENT(12) {ASYNCHRONOUS|error1}";
+    private static final String DEFAULT_INVALID_FOUR = "RUN {ACQACTIVITY|error1}";
+
+    private static final String TRANSID_ALL = "RUN TRANSID(12) CHANNEL(123) CHILD(1)";
```

Comment:
```
DFH7226I           E        00001         SUB-OPERAND '1' OF 'CHILD' KEYWORD SHOULD BE A DATA REFERENCE NOT AN
                                          EXPRESSION OR A CONSTANT. COMMAND NOT TRANSLATED.                   
```

Reply:
Correct, for a real run we would need something else in place of a 1 there. However, the unit test is testing the validity of the keyword token elements, not the arguments.

I suppose a more technically appropriate approach would be to use mocks to signify that we don't actually care about what variables are passed in, but considering any sort of string will do from a parser perspective, it seems like a bit of overkill.